src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java

```diff
@@ -127,30 +140,33 @@ private void init() {
     public int executeCollection() {
         List<Collector> activeCollectors = getActiveCollectors(parameterMap);
 
+        if (activeCollectors.isEmpty()) {
+            logger.info("No collectors are active. Nothing will be collected!");
+            return 0;
+        }
+
         int result = 0;
-        if (!activeCollectors.isEmpty()) {
-            for (Collector collector : activeCollectors) {
-                collector.setParams(parameterMap);
-                result = collector.collect();
-                if (result != 0) {
-                    return result;
-                }
+        for (Collector collector : activeCollectors) {
+            collector.setParams(parameterMap);
+            result = collector.collect();
+            if (result != 0) {
+                return result;
             }
         }
 
-        File file = new File((String) parameterMap.get(ParamConstants.DIR_PARAM));
+        Path filePath = Paths.get((String) parameterMap.get(DIR_PARAM));
 
-        try (FileOutputStream fileOutputStream = new FileOutputStream(file.getAbsolutePath() + ".zip")) {
-            ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream);
-            compressDirectory(file, file.getName(), zipOutputStream);
+        ZipOutputStream zipOutputStream = null;
+        try (FileOutputStream fileOutputStream = new FileOutputStream(filePath.toFile() + ".zip")) {
+            zipOutputStream = new ZipOutputStream(fileOutputStream);
```

Comment:
I don't see a reason for this variable to be declared outside of the try block
```suggestion
        try (FileOutputStream fileOutputStream = new FileOutputStream(filePath.toFile() + ".zip")) {
            ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream);
```

Reply:
resolved in #21